pkg/operator/starter.go

```diff
@@ -208,6 +208,17 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 	var notOnSingleReplicaTopology resourceapply.ConditionalFunction = func() bool {
 		return infrastructure.Status.ControlPlaneTopology != configv1.SingleReplicaTopologyMode
 	}
+	isDelayedDualReplica := false
+	if infrastructure.Status.ControlPlaneTopology == configv1.DualReplicaTopologyMode {
+		etcdNamespace, err := kubeClient.CoreV1().Namespaces().Get(ctx, "openshift-etcd", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		_, isDelayedDualReplica = etcdNamespace.Annotations["openshift.io/delayed-bootstrap"]
+	}
+	var requireMultipleEtcdEndpoints resourceapply.ConditionalFunction = func() bool {
+		return !(infrastructure.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode || isDelayedDualReplica)
+	}
```

Comment:
I'm not happy about `!(infrastructure.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode || isDelayedDualReplica)`, lets try this
```suggestion
	isMultipleEtcdEndpointsRequired := true
	switch infrastructure.Status.ControlPlaneTopology {
	case configv1.SingleReplicaTopologyMode:
		isMultipleEtcdEndpointsRequired = false
	case configv1.DualReplicaTopologyMode:
		etcdNamespace, err := kubeClient.CoreV1().Namespaces().Get(ctx, "openshift-etcd", metav1.GetOptions{})
		if err != nil {
			return err
		}
		_, isMultipleEtcdEndpointsRequired = etcdNamespace.Annotations["openshift.io/delayed-bootstrap"]
	}
	var requireMultipleEtcdEndpoints resourceapply.ConditionalFunction = func() bool {
		return isMultipleEtcdEndpointsRequired
	}
```